### ISSUE TITLE
end-to-end example as a notebook?

### ISSUE THREAD

**Issue by vgoklani** [damitkwr/ESRNN-GPU#13]
Any chance you could push an end-to-end example as a jupyter notebook? It's really hard to follow the codebase... thanks!

**Comment by damitkwr** [comment]
Sure, we will start on getting a notebook example out

**Comment by andmib** [comment]
I'm excited to see the end-to-end example. I'd like to generalize this to non-M4 time series, and that'll be super helpful for it. Thanks!

**Comment by vgoklani** [comment]
Thanks @damitkwr

**Comment by stevevaius77** [comment]
Really looking forward to it. Thnx for efforts by the way

**Comment by andrewczgithub** [comment]
Thank you @damitkwr looking forward to it :).

Many thanks,
Andrew

**Comment by 12suyash** [comment]
Me also looking forward for it

**Comment by cbpaw** [comment]
> I'm excited to see the end-to-end example. I'd like to generalize this to non-M4 time series, and that'll be super helpful for it. Thanks!

Very interested in this

**Comment by andrewczgithub** [comment]
Hi all!
Was wondering if there was an update on the above

**Comment by misaelmoura** [comment]
Would be good a generalization for different data and a way to get the forecasted values.

**Comment by Henry-Lidgley** [comment]
Would also be very grateful for any further documentation for the code beyond Smyl's Uber page and the paper. I've come from using Keras and finding it difficult to follow. It's training fine but would like to generalise on non-M4 data (though also not sure how to obtain forecasts). Even restricting the M4 data to a fewer number of time series breaks the code due to a batch size issue in Pytorch. Will keep trying to get my head around it. Many thanks.

**Comment by firmai** [comment]
Why has nothing been done to this end? Some smart person somewhere should have a colab they can share.

**Comment by pluto7liyu** [comment]
> Sure, we will start on getting a notebook example out

Hi, Sir, do you have update on this one. Very interested in learning this approach!

**Comment by Henry-Lidgley** [comment]
For those interested in ES-RNN implement also check out N-BEATs: 
https://towardsdatascience.com/n-beats-beating-statistical-models-with-neural-nets-28a4ba4a4de8 
https://github.com/philipperemy/n-beats/tree/master/examples

**Comment by samuelgneff** [comment]
Currently working on a colab notebook. Hoping to extend the implementation to non-M4 data where you may have additional metadata (day of the week, holidays, etc.). Happy to share once it's done.

**Comment by firmai** [comment]
Looking forward to it.

**Comment by FlorisCalkoen** [comment]
@samuelgneff How is your project going :)? Looking forward to your nb! Cheers

**Comment by FlorisCalkoen** [comment]
Please find a simple Colab NB example here: https://gist.github.com/florisrc/e9bc4cb054c6ad1c8f1ac43a9c21d09f

**Comment by dan-r95** [comment]
> Please find a simple Colab NB example here: https://gist.github.com/florisrc/e9bc4cb054c6ad1c8f1ac43a9c21d09f

I added tensorflow 1 compatibility and a direct link to 
[ google colab here](https://colab.research.google.com/drive/18GJ9fbnKuB8sNT0Y1L_MoGVMfNnNjLVG?usp=sharing) :+1: .

**Comment by caugusta** [comment]
How long should we expect this code to run for?

**Comment by FangSimon** [comment]
> I'm excited to see the end-to-end example. I'd like to generalize this to non-M4 time series, and that'll be super helpful for it. Thanks!

Has anyone successfully generalized this to non-m4 time series and has a notebook of this? For example, the M3 time series?
